posts/views.py

```diff
@@ -215,6 +220,168 @@ def post(self, request, pk, *args, **kwargs):
 
+@method_decorator(login_required, name='dispatch')
+# def selectPerson(request):
+#     author_list = Author.objects.all()
+#     context = {
+#             'author_list':author_list,
+#         }
+#     return render(request, 'selectPerson.html', context)
+class selectPersonView(View):
+    def get(self, request, *args, **kwargs):
+        form = PostForm()
+        share_form = ShareForm()
+        author_list = Author.objects.all()
+        context = {
+            'form': form,
+            'author_list': author_list,
+            # 'source_post': source_post,
+            # 'original_post': original_post,
+        }
+        return render(request, 'selectPerson.html', context)
+
+    def post(self, request, *args, **kwargs):
+        # form = PostForm(request.POST, request.FILES)
+        # author_list = Author.objects.all()
+        username = request.POST.get('specificUserName', '')
+        # print(username)
+        try:
+            selected_author = Author.objects.get(displayName = username)
+        except:
+            selected_author = None
+
+        try:
+            post = Post.objects.filter(author__username=request.user.username).order_by('-published').first()
+            # print(Post.objects.filter(author__username=request.user.username))
+            # print(post)
+        except:
+            post = []
+
+        # check if they are true friend.
+        user = Author.objects.get(username=request.user.username)
+        try:
+            myfollowers = Followers.objects.filter(
+                user__username=user.username).first().items.all()
+        except:
+            myfollowers = []
+        if selected_author in myfollowers:
+            # if the host is local
+            if selected_author.host in localHostList:
+                try:
+                    hisfollowers = Followers.objects.filter(
+                                user__username=selected_author.username).first().items.all()
+                except:
+                    hisfollowers = []
+                if user in hisfollowers:
+                    # in this case you are true friends
+                    try:
+                        # Inbox.objects.filter(author__username=selected_author_info.username).first().items.add(post)
+                        # add the new post to my own inbox
+                        InboxItem.objects.create(
+                            inbox=Inbox.objects.filter(
+                                author__username=selected_author.username).first(),
+                            inbox_item_type="post",
+                            item=post,
+                        )
+                        try:
+                            InboxItem.objects.create(
+                                inbox=Inbox.objects.filter(
+                                    author__username=request.user.username).first(),
+                                inbox_item_type="post",
+                                item=post,
+                            )
```

Comment:
nested `try except` blocks arent needed. if it fails anywhere in the try block it will be caught be except block. this can be revised for cleaner code.